components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -4025,9 +4026,15 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             "" /* empty search term returns any */,
         );
         const hasAnyOrg = rows.length > 0;
-
+        let isCompleted = false;
+        for (const row of rows) {
+            isCompleted = await this.teamDB.hasActiveSSO(row.id);
+            if (isCompleted) {
+                break;
+            }
+        }
```

Comment:
I wonder if we should instead update the regular Team object response to include whether it has SSO enabled or not, and get it from there instead of doing a one-off check. This would mean we join the `d_b_team` table with the `d_b_oidc_client_configs` table to determine that, but that's an implementation detail of the DB layer, and the service layer here would just consume it as a property of the team.

This could be exposed in the Team object on both server, and public-api, which would then give you a better check in the implementation of Get/List Team for whether you should lookup an invite, or not

Reply:
Both could work. The approach in this PR is a smaller change and given that the information is only currently needed in rarely used places (around invites) asking there seems less costly than fetching orgs with joins every time.